### ISSUE TITLE
dpkg-buildpackage fails, missing debian/grr/DEBIAN/control

### ISSUE THREAD

**Issue by zbuc** [google/grr#423]
Trying to run `grr_config_updater repack_clients` from an Amazon Linux box that I installed the `dpkg` and `debhelper` packages on from source:

```
Using configuration <GrrConfigManager  file="/home/grr_user/GRR_NEW/install_data/etc/grr-server.yaml"  file="/home/grr_user/GRR_NEW/install_data/etc/server.local.yaml" >
Repacking template: /home/grr_user/GRR_NEW/grr-response-templates/templates/grr_3.1.0.2_amd64.deb.zip
Using context: ['ClientBuilder Context', 'Target:LinuxDeb', 'Platform:Linux', 'Target:Linux', 'Arch:amd64', 'ClientBuilder Context', 'Target:Linux'] and labels: []
ERROR:2016-08-24 18:07:35,347 build:703] Error calling ['/usr/bin/dpkg-buildpackage', '-uc', '-d', '-b', u'-aamd64'].
ERROR:2016-08-24 18:07:35,347 build:704] dpkg-buildpackage: info: source package grr
dpkg-buildpackage: info: source version 3102-1
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by GRR <grr-dev@googlegroups.com>
dpkg-architecture: warning: specified GNU system type x86_64-linux-gnu does not match CC system type x86_64-amazon-linux, try setting a correct CC environment variable
dpkg-buildpackage: warning: debian/rules is not executable; fixing that
 dpkg-source --before-build dist
dpkg-buildpackage: info: host architecture amd64
 fakeroot debian/rules clean
dh clean
   debian/rules override_dh_clean
make[1]: Entering directory `/tmp/tmpcaUUSR/dist'
# Nothing to do here, directory is already wiped and set up.
make[1]: Leaving directory `/tmp/tmpcaUUSR/dist'
 debian/rules build
dh build
 fakeroot debian/rules binary
dh binary
   debian/rules override_dh_prep
make[1]: Entering directory `/tmp/tmpcaUUSR/dist'
# Nothing to do here, directory is already wiped and set up.
make[1]: Leaving directory `/tmp/tmpcaUUSR/dist'
   debian/rules override_dh_link
make[1]: Entering directory `/tmp/tmpcaUUSR/dist'
dh_link "/usr/lib/grr/grr_3.1.1.0_amd64/grrd" "/usr/sbin/grrd"
make[1]: Leaving directory `/tmp/tmpcaUUSR/dist'
   debian/rules override_dh_strip
make[1]: Entering directory `/tmp/tmpcaUUSR/dist'
# Necessary otherwise the PyInstaller build executables will not run
make[1]: Leaving directory `/tmp/tmpcaUUSR/dist'
   debian/rules override_dh_makeshlibs
make[1]: Entering directory `/tmp/tmpcaUUSR/dist'
# These add a lot of unnecessary dependencies to the deb package because
# it just looks for .so files and adds them as dependencies. Since we
# are shipping our .so's with Pyinstaller this is unnecessary
make[1]: Leaving directory `/tmp/tmpcaUUSR/dist'
   debian/rules override_dh_shlibdeps
make[1]: Entering directory `/tmp/tmpcaUUSR/dist'
# These add a lot of unnecessary dependencies to the deb package because
# it just looks for .so files and adds them as dependencies. Since we
# are shipping our .so's with Pyinstaller this is unnecessary
make[1]: Leaving directory `/tmp/tmpcaUUSR/dist'
   debian/rules override_dh_builddeb
make[1]: Entering directory `/tmp/tmpcaUUSR/dist'
# Older distros don't support xz so we use gzip instead.
dh_builddeb -- -Zgzip
dpkg-deb: error: failed to open package info file 'debian/grr/DEBIAN/control' for reading: No such file or directory
dh_builddeb: dpkg-deb -Zgzip --build debian/grr .. returned exit code 2
make[1]: *** [override_dh_builddeb] Error 1
make[1]: Leaving directory `/tmp/tmpcaUUSR/dist'
make: *** [binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary gave error exit status 2

Repacking template /home/grr_user/GRR_NEW/grr-response-templates/templates/grr_3.1.0.2_amd64.deb.zip failed: Command '['/usr/bin/dpkg-buildpackage', '-uc', '-d', '-b', u'-aamd64']' returned non-zero exit status 2
Failed to repack /home/grr_user/GRR_NEW/grr-response-templates/templates/grr_3.1.0.2_amd64.deb.zip.
```

Is it due to the fakeroot, or something else? Has anyone successfully used GRR to create `.deb` packages on Amazon Linux or CentOS?

**Comment by destijl** [comment]
I've seen people have problems where /tmp is noexec:
https://github.com/google/grr/issues/417

although this is a fairly different looking error. You should set a breakpoint here and try to make the dpkg call manually, since that seems to be what's failing:
https://github.com/google/grr/blob/master/grr/lib/build.py#L701

it works for me on ubuntu, and in the travis ubuntu environment.

**Comment by zbuc** [comment]
The specific error is that it can't find that package info file:

`dpkg-deb: error: failed to open package info file 'debian/grr/DEBIAN/control' for reading: No such file or directory`

It sounds like it should be at `/tmp/tmpcaUUSR/dist/debian/grr/DEBIAN/control`.

What is supposed to create this file?

**Comment by zbuc** [comment]
This appears to be an issue with the Debian build tools when they're installed on Amazon Linux, not a GRR bug (though frustrating since I'm running into it while deploying GRR).

I tried a couple things:

From the working directory `/tmp/tmpg20WYJ/dist`, rename the `debian` directory to `DEBIAN` and execute `dpkg-deb -Zgzip --build . ..` (instead of the original command `dpkg-deb -Zgzip --build debian/grr ..`).

This results in a couple syntax issues with the `DEBIAN/control` file -- it didn't like the white space or the lack of a `Version:` line, so I added those, and now am receiving errors about the flags passed to `tar`:

```
$ dpkg-deb -Zgzip --build . ..
dpkg-deb: building package 'grr' in '../grr_1_any.deb'.
tar: unrecognized option '--clamp-mtime'
Try `tar --help' or `tar --usage' for more information.
dpkg-deb: error: subprocess tar -cf returned error exit status 64
```

I am going to proceed by comparing the .deb packaging to a GRR install on Ubuntu. I don't think there's anything GRR needs to do, unless you're interested in whatever special-case modifications I wind up making to support Amazon Linux.
